src/concrete/ml/sklearn/neighbors.py

```diff
@@ -0,0 +1,117 @@
+"""Implement sklearn linear model."""
+from typing import Any, Dict
+
+import sklearn.linear_model
+
+from .base import SklearnKNeighborsClassifierMixin
+
+
+# pylint: disable=invalid-name,too-many-instance-attributes
+class KNeighborsClassifier(SklearnKNeighborsClassifierMixin):
+    """A k-nearest classifier model with FHE.
+
+    Parameters:
+        n_bits (int, Dict[str, int]): Number of bits to quantize the model. If an int is passed
+            for n_bits, the value will be used for quantizing inputs and weights. If a dict is
+            passed, then it should contain "op_inputs" and "op_weights" as keys with
+            corresponding number of quantization bits so that:
+            - op_inputs : number of bits to quantize the input values
+            - op_weights: number of bits to quantize the learned parameters
+            Default to 8.
+
+    For more details on KNeighborsClassifier please refer to the scikit-learn documentation:
+    https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html
+    """
+
+    sklearn_model_class = sklearn.neighbors.KNeighborsClassifier
+    _is_a_public_cml_model = True
+
+    def __init__(
+        self,
+        n_bits=2,
+        n_neighbors=3,
+        *,
+        weights="uniform",
+        algorithm="auto",
+        leaf_size=30,
+        p=2,
+        metric="minkowski",
+        metric_params=None,
+        n_jobs=None,
+    ):
+        # Call SklearnKNeighborsClassifierMixin's __init__ method
+        super().__init__(n_bits=n_bits)
+
+        self.n_neighbors = n_neighbors
+        self.algorithm = algorithm
```

Comment:
also need to assert algorithm is "brute force" or "auto". the other ones (k-d tree, ball tree) can not be done in FHE. see https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html